wz_strategic_plugin/src/wz_strategic_plugin.cpp

```diff
@@ -0,0 +1,282 @@
+
+        wml_.reset(new carma_wm::WMListener());
+        // set world model point form wm listener
+        wm_ = wml_->getWorldModel();
+
+        discovery_pub_timer_ = pnh_->createTimer(
+            ros::Duration(ros::Rate(10.0)),
+            [this](const auto &) { plugin_discovery_pub_.publish(plugin_discovery_msg_); });
+
+    }
+
+    void WzStrategicPlugin::pose_cb(const geometry_msgs::PoseStampedConstPtr &msg)
+    {
+        pose_msg_ = msg;
+        lanelet::BasicPoint2d curr_loc(pose_msg_->pose.position.x, pose_msg_->pose.position.y);
+        current_loc_ = curr_loc;
+    }
+
+    void WzStrategicPlugin::twist_cb(const geometry_msgs::TwistStampedConstPtr &msg)
+    {
+        current_speed_ = msg->twist.linear.x;
+    }
+
+    std::string WzStrategicPlugin::traffic_light_interpreter(lanelet::CarmaTrafficLightState& state)
+    {
+        std::string traffic_light_inter;
+        switch(state) {
+            case lanelet::CarmaTrafficLightState::UNAVAILABLE:
+                traffic_light_inter = "R";
+                break;
+            case lanelet::CarmaTrafficLightState::DARK:
+                traffic_light_inter = "R";
+                break;
+            case lanelet::CarmaTrafficLightState::STOP_THEN_PROCEED:
+                traffic_light_inter = "R";
+                break;
+            case lanelet::CarmaTrafficLightState::STOP_AND_REMAIN:
+                traffic_light_inter = "R";
+                break;
+            case lanelet::CarmaTrafficLightState::PRE_MOVEMENT:
+                traffic_light_inter = "R";
+                break;
+            case lanelet::CarmaTrafficLightState::PERMISSIVE_MOVEMENT_ALLOWED:
+                traffic_light_inter = "R";
+                break;
+            case lanelet::CarmaTrafficLightState::PROTECTED_MOVEMENT_ALLOWED:
+                traffic_light_inter = "R";
+                break;
+            case lanelet::CarmaTrafficLightState::PERMISSIVE_CLEARANCE:
+                traffic_light_inter = "R";
+                break;
+            case lanelet::CarmaTrafficLightState::PROTECTED_CLEARANCE:
+                traffic_light_inter = "R";
+                break;
+            case lanelet::CarmaTrafficLightState::CAUTION_CONFLICTING_TRAFFIC:
+                traffic_light_inter = "R";
+                break;
+        }
+
+        return traffic_light_inter;
+
+    }
+
+    bool WzStrategicPlugin::planManeuverCb(cav_srvs::PlanManeuversRequest &req, cav_srvs::PlanManeuversResponse &resp)
+    {
+        ROS_DEBUG("Finding car infomrmation");
+
+        // find car_down_track
+        double current_car_down_track = wm_->routeTrackPos(current_loc_).downtrack;
+        ROS_DEBUG("current_car_down_track %d", current_car_down_track);
+
+        auto current_lanelets = lanelet::geometry::findNearest(wm_->getMap()->laneletLayer, current_loc_, 1);
+        auto current_lanelet = current_lanelets[0];
```

Comment:
There is a check in the old implementation for route following that ensures that the current lanelet is on the shortest path. There is a chance here that the current lanelet picked up is not one in the route, i think it would be useful to have that check